pyfr/solvers/base/system.py

```diff
@@ -248,7 +261,7 @@ def _prepare_kernels(self, t, uinbank, foutbank):
         _, binders = self._get_kernels(uinbank, foutbank)
 
         for b in self._bc_inters:
-            b.prepare(t)
+            b.prepare(t, self, uinbank)
```

Comment:
How is the bank used?  The system normally tries to hide these details

Reply:
I was using `system.ele_scal_upts(uinbank)` to get the current solution to do the mass flow integration on. I could just use a fixed bank in the BC like `system.ele_scal_upts(0)` as the difference between the solution at a different RK stage/previous time step would have negligible effect on the mass flow controller (which checks out on a simple test case). Or I could pass the solution to prepare, i.e. `b.prepare(self, t, kerns, system.ele_scal_upts(uinbank))`?

Reply:
I think the current approach is fine if a bit inefficient (as if multiple BCs all want the same thing we'll end up copying it multiple times).  With that said the long term solution is to migrate the surface quadrature to the backend so it is fine to keep things as they are for the moment (this being a much better solution as it avoids ever needing to copy the solution).